ftl/python/layer_builder.py

```diff
@@ -14,54 +14,227 @@
 """This package implements the Python package layer builder."""
 
 import datetime
+import logging
 import os
 import subprocess
-import logging
+import tempfile
 
 from ftl.common import ftl_util
 from ftl.common import single_layer_image
 from ftl.common import tar_to_dockerimage
 
 
 class PackageLayerBuilder(single_layer_image.CacheableLayerBuilder):
-    def __init__(self, ctx, descriptor_files, pkg_dir, dep_img_lyr):
+    def __init__(self, ctx=None, descriptor_files=None,
+                 pkg_dir=None, dep_img_lyr=None, cache=None):
         super(PackageLayerBuilder, self).__init__()
         self._ctx = ctx
         self._pkg_dir = pkg_dir
         self._descriptor_files = descriptor_files
         self._dep_img_lyr = dep_img_lyr
+        self._cache = cache
```

Comment:
what is the rationale for moving the caching down into into the layer builders?

Reply:
the cache logging requires information about the packages being installed in the layers, which is only available in the layer builders themselves. so the options were to return package information back up to the builder, or to move the caching down into the layer builders. since that's where the actual layers are being constructed, the second option made more sense to me.